frontend/packages/console-plugin-sdk/src/utils/useTranslatedExtensions.ts

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { TFunction } from 'i18next';
+import { Extension, LoadedExtension } from '@console/dynamic-plugin-sdk/src/types';
+import { isTranslatableString, translateExtensionDeep } from './extension-i18n';
+import useTranslationExt from './useTranslationExt';
+
+/**
+ * `translateExtensionDeep` mutates the extension for translations. We need to store a
+ * semi-permanent mapping of the translation keys values.
+ *
+ * Structured as: { [extension.UID]: { [propertyPathToTranslation]: translationKey } }
+ */
+const translationKeyMap: Record<string, Record<string, string>> = {};
+
+const useTranslatedExtensions = <E extends Extension>(
+  extensions: LoadedExtension<E>[],
+): LoadedExtension<E>[] => {
+  const { t } = useTranslationExt();
+  // TODO: Drop ref and start using useEffect when we are not mutating extensions
+  const lastTRef = React.useRef<TFunction>(null);
+
+  if (t !== lastTRef.current) {
+    extensions.forEach((e) => {
+      const UID = e.uid;
+      translateExtensionDeep(
+        e,
+        (value, path): value is string => {
```

Comment:
Nit: the `value is string` part can be removed, it's auto-inferred from `value` type.

Reply:
`value` is not a string... it's a type guard and it is typed as `unknown` (was `any`). See the original implementation that is just propagated forward to here: https://github.com/openshift/console/pull/11278/files#diff-dac0a51cba2153e1d739813ca64b2585438bb477e7e83dc6a096fb2ad371f1bcL10

Reply:
Huh, I'd expect the auto-inference to kick in since `translateExtensionDeep` second argument is typed as `PredicateCheck<string>`. Thanks for the info.

Reply:
The error is:
```
TS2345: Argument of type '(value: unknown, path: string) => boolean' is not assignable to parameter of type 'PredicateCheck<string>'.   Signature '(value: unknown, path: string): boolean' must be a type predicate.
```

Odd it does not allow the inference through that type... but it is what it is. Must be a compiler choice to not infer predicates.